### PR TITLE
Fixes for PMI1/2 support

### PR DESCRIPTION
**Pull request by igor-ivanov**
@rhc54, @artpol84 could you look at set of fixes related PMI1/2
It also includes pmi test.

### CODE REVIEW

**Comment by mellanox-github** [comment]
* Coverity found 6 errors for pmix
* Coverity report at http://bgate.mellanox.com/jenkins/job/gh-ompi-pmix-pr//81/Coverity_Report <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by rhc54** [comment]
:+1:

File: src/client/pmi1.c [hunk 1]
```diff
@@ -167,47 +221,49 @@ int PMI_KVS_Get( const char kvsname[], const char key[], char value[], int lengt
...
+        PMIX_INFO_LOAD(info, PMIX_COLLECT_DATA, &val, PMIX_BOOL);
         ninfo = 1;
     }
     rc = PMIx_Fence(NULL, 0, info, ninfo);
 
-    if( NULL != info ){
+    if (NULL != info) {
         PMIX_INFO_DESTRUCT(info);
     }
     return rc;
```
Comment (artpol84) [inline src/client/pmi1.c]: We need conver_error here.

File: src/client/pmi2.c [hunk 2]
```diff
@@ -37,15 +37,23 @@
...
+        if (!pmi2_init) {     \
+            return PMI2_FAIL; \
+        }                    \
+    } while (0)
+
 /* local functions */
 static pmix_status_t convert_int(int *value, pmix_value_t *kv);
 static int convert_err(pmix_status_t rc);
 static pmix_proc_t myproc;
+static int pmi2_init = 0;
```
Comment (artpol84) [inline src/client/pmi2.c]: Can we use PMIx_Initialized() instead?

**Comment by artpol84** [comment]
All looks good (although it's little bit late and the PR is merged :).

**Pull request merged**